django_prices/forms.py

```diff
@@ -1,49 +1,69 @@
 import itertools
+from typing import List, Optional, Tuple
 
+from babel.numbers import get_currency_symbol
 from django import forms
+from django.core.validators import ValidationError
 from prices import Money
 
-from .validators import (
-    MaxMoneyValidator,
-    MinMoneyValidator,
-    MoneyPrecisionValidator,
-    ValidationError,
-)
+from .validators import MaxMoneyValidator, MinMoneyValidator, MoneyPrecisionValidator
 from .widgets import FixedCurrencyMoneyInput, MoneyInput
+from .templatetags.prices_i18n import get_locale_data
 
 __all__ = ("MoneyField", "MoneyInput")
 
 
+def _get_currency_choices(currencies: List[str]) -> List[Tuple[str, str]]:
+    """Generate choices for SelectField.
+    As a label we are presenting currency symbol."""
+    _, locale_code = get_locale_data()
+    currency_choices = [
+        (code, get_currency_symbol(code, locale_code)) for code in currencies
+    ]
+    return currency_choices
+
+
 class MoneyField(forms.MultiValueField):
+    """A field for Money objects.
+
+    When only one currency is provided, it renders as FixedCurrencyMoneyInput,
+    when more you will get MoneyInput which allow you to choose both amount
+    and currency.
+
+    `available_currencies` argument is expected to be list of currency codes."""
+
     def __init__(
         self,
-        available_currencies,
-        widget=MoneyInput,
-        max_values=None,
-        min_values=None,
-        max_digits=None,
-        decimal_places=None,
+        available_currencies: List[str],
+        widget=None,
+        max_values: Optional[List] = None,
+        min_values: Optional[List] = None,
+        max_digits: int = None,
+        decimal_places: int = None,
         validators=(),
         *args,
         **kwargs
     ):
         decimal_field = forms.DecimalField(
             max_digits=max_digits, decimal_places=decimal_places
         )
-        fields = (decimal_field, forms.ChoiceField(choices=available_currencies))
-        if isinstance(widget, type):
-            widget_instance = widget(available_currencies)
-            if (
-                isinstance(widget_instance, MoneyInput)
-                and len(available_currencies) <= 1
-            ):
-                widget_instance = FixedCurrencyMoneyInput(
-                    currency=available_currencies[0]
-                    if len(available_currencies) == 1
-                    else "",
-                    attrs={"type": "number", "step": "any"},
-                )
-                fields = (decimal_field, forms.CharField())
+
+        choices = _get_currency_choices(available_currencies)
+        fields = (decimal_field, forms.ChoiceField(choices=choices))
+
+        if widget is not None:
+            raise NotImplementedError()
```

Comment:
Proper error message about why initializing `MoneyField ` with `widget` is an error would potentially make this easier to debug for developers unfamiliar with changes..